documentation/development/docker.md

```diff
@@ -6,54 +6,33 @@ This describes how to create an environment for development using Docker.
 
 ## Running OCIM in a docker container
 
-First, set your `.env` file as you would normally:
-```env
-DEBUG=True
-OPENSTACK_USER='username'
-OPENSTACK_PASSWORD='password'
-OPENSTACK_TENANT='tenant-name'
-OPENSTACK_AUTH_URL='https://auth.cloud.ovh.net/v2.0'
-OPENSTACK_REGION='BHS1'
-OPENSTACK_SANDBOX_SSH_KEYNAME='keypair-name'
-DEFAULT_INSTANCE_BASE_DOMAIN='example.com'
-GANDI_API_KEY='api-key'
-GITHUB_ACCESS_TOKEN='github-token'
-SECRET_KEY='tests'
-DEFAULT_INSTANCE_MYSQL_URL=...
-DEFAULT_RABBITMQ_API_URL=...
-DEFAULT_INSTANCE_RABBITMQ_URL=...
-DEFAULT_MONGO_REPLICA_SET_USER=...
-DEFAULT_MONGO_REPLICA_SET_PASSWORD=...
-DEFAULT_MONGO_REPLICA_SET_NAME=...
-DEFAULT_MONGO_REPLICA_SET_PRIMARY=...
-DEFAULT_MONGO_REPLICA_SET_HOSTS=...
-REDIS_URL=...
-```
+First, set your `.env` file as you would normally,
+but set the following vars:
 
-Start a one-off container for setup of the backend:
 ```sh
-# --rm destroys the container at the end
-docker-compose run --rm ocim bash
+ALLOWED_HOSTS='["*"]'
+DATABASE_URL='postgres://opencraft@postgresql/opencraft'
+REDIS_URL='redis://redis:6379'
+```
 
-# then, inside the container...
+Run some initial setup tasks in the backend:
 
-# initialize datastores
-make migrate
-# create super user
-make manage createsuperuser
-# exit the container
-exit
+```sh
+docker-compose run --rm ocim make migrate
+docker-compose run --rm ocim make manage createsuperuser
 ```
 
-Then, start another one-off container for setting up the client-facing frontend:
+And on the frontend:
+
 ```sh
 docker-compose run --rm ocim-frontend bash -c 'npm run build-api-client && npm install'
 ```
 
 Then, start OCIM:
+
 ```sh
-docker-compose up
+docker-compose up -d
```

Comment:
Out of curiosity, what's the rationale for running detached here?

To clarify, I can see why someone would want to run it detached; however, I prefer to assume the reader might not know `docker-compose`  well and thus be confused when he "starts OCIM" but doesn't see any output.

Reply:
> be confused when he "starts OCIM" but doesn't see any output.

@kewne Actually, it would see the message that the containers are started, though it is also true that the logs wouldn't be shown. We could add `docker-compose up -d && docker-compose logs` as a separate command in the docs to have the detached behavior with showing the logs too.

Reply:
@kewne,

> what's the rationale for running detached here?

That is the default behavior of the edx devstack and others I've used in the past, with separate commands for viewing logs when you need them.  I second @swalladge's implicit opinion that this is a generally better default than the alternative.

Reply:
@gabor-boros 

My point isn't that plain `docker-compose up` is the best command for every or even most workflows but rather that it is the least surprising behavior, especially considering that someone reading this might not be an expert in Docker and Docker Compose;
conversely, someone who *is* an expert doesn't need additional documentation to know he *can* use `-d`.
The intention here isn't to impose or suggest a particular workflow, it's literally "this is the **simplest** sequence of commands that gets you started": I find it hard to argue that `docker-compose up -d && docker-compose logs` is superior to `docker-compose` in that regard.

> That is the default behavior of the edx devstack and others I've used in the past

@arbrandes I'm a bit surprised by your reply: I've always considered starting stuff in the foreground to be the "correct" behavior, so much so that it's the default for pretty much everything (`docker`, `./manage runserver`, etc).
Besides, this is actually something I dislike about the way devstack works: it attempts to hide its use of `docker-compose` but, in doing so, creates a bespoke interface that won't allow me, an experience user, to properly customize commands.

For the sake of "meeting in the middle", perhaps the best would be to state this uses Docker Compose, and link to its documentation.
Optionally, we could document "useful beginner commands", in which case I feel `-d` and `logs` would be very appropriate.

Reply:
> I've always considered starting stuff in the foreground to be the "correct" behavior, so much so that it's the default for pretty much everything

As far as Unix-like systems go, you're right, I agree, and it's why `-d` is in fact an optional flag.  No argument there.

It's just that for development stacks that feature more than one process, I find launching them in the background and looking for individual logs when I need them to be the more efficient experience.  In Ocim's case it's harder to press this point, as it is evidently a single process.  So I won't. :)

Reply:
whoa whoa this is a work in progress dump of notes I was making while trying to get everything working! :sweat_smile: Let's not bikeshed here just yet. :stuck_out_tongue: 


>  In Ocim's case it's harder to press this point, as it is evidently a single process. 

Ocim is no longer a single process; there's a separate frontend app too.  And the docker compose launches postgres, redis, etc.

I'll come back to this discussion once I've had a chance to work with it a bit longer.

Reply:
Ok, I updated the docs so it's clear what's happening.  Also added a note about viewing the logs.

Reply:
> perhaps the best would be to state this uses Docker Compose, and link to its documentation.

The docs literally show use of the `docker-compose` command, so nothing is being hidden. :thinking: